chainer/serializers/npz.py

```diff
@@ -178,6 +178,8 @@ def __call__(self, key, value):
             value_type = type(value)
             dataset_arr = numpy.asarray(dataset)
             if (issubclass(dataset_arr.dtype.type, numpy.number)
+                    and not (issubclass(dataset_arr.dtype.type, numpy.integer)
+                             and value_type is int)
```

Comment:
Could you add a brief comment on the issue?  I'd like to see the following

- At least, Python scalars and NumPy scalars are supported.
- The `to` arg of `numpy.can_cast` is treated as a dtype spec.
- `numpy.dtype(int) == numpy.int32` in some environments.